src/js/util/settings.js

```diff
@@ -60,3 +60,9 @@ exports.getSetting = function(userBlob, settingName, def) {
   }
   return def;
 };
+
+exports.getClearServers = function(servers) {
+  return _.map(servers, function(server) {
+    return _.pick(server, 'host', 'port', 'secure');

```

Comment:
$$hashKey added by angular. Same object used as parameter to Remote, so url property get added by Server.
Also new servers added from interface has isEmptyServer property.